kube/file-vault/file-vault-deployment.yml

```diff
@@ -0,0 +1,233 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  {{ if eq .KUBE_NAMESPACE .PROD_ENV }}
+  name: file-vault
+  {{ else if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+  annotations:
+    downscaler/uptime: {{ .NON_PROD_AVAILABILITY }}
+  name: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+  {{ else }}
+  annotations:
+    downscaler/uptime: {{ .NON_PROD_AVAILABILITY }}
+  name: file-vault
+  {{ end }}
+spec:
+  selector:
+    matchLabels:
+      {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+      name: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+      service: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+      {{ else }}
+      name: file-vault
+      service: file-vault
+      {{ end }}
+  template:
+    metadata:
+      labels:
+        {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+        name: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+        service: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+        {{ else }}
+        name: file-vault
+        service: file-vault
+        {{ end }}
+    spec:
+      containers:
+        - name: file-vault
```

Comment:
I don't believe the first setup should have filevault, it makes it more complicated

Reply:
I have it configured, as it is part of the sub-tasks...unless we separate it as a main ticket on its own?

Reply:
yeah
you want to do it step by step imho